recommendation/src/main/java/com/yas/recommendation/topology/ProductTopology.java

```diff
@@ -0,0 +1,359 @@
+import org.apache.kafka.streams.kstream.Consumed;
+import org.apache.kafka.streams.kstream.KTable;
+import org.apache.kafka.streams.kstream.Produced;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * ProductTopology defines a Kafka Streams topology that processes product-related data,
+ * joins it with brand, category, and attribute information, and aggregates details
+ * for enriched product output. This topology generates various materialized views for
+ * different stages of data enrichment and outputs the final processed product details
+ * to a Kafka sink topic.
+ */
+@Component
+@Slf4j
+public class ProductTopology extends AbstractTopology {
+
+    private static final String PRODUCT_MATERIALIZED_VIEW = "product-mv";
+    private static final String BRAND_MATERIALIZED_VIEW = "brand-mv";
+    private static final String PRODUCT_BRAND_MATERIALIZED_VIEW = "product-brand-mv";
+    private static final String CATEGORY_MATERIALIZED_VIEW = "category-mv";
+    private static final String PRODUCT_CATEGORY_MATERIALIZED_VIEW = "product-category-mv";
+    private static final String PRODUCT_CATEGORY_NON_NULL_MATERIALIZED_VIEW = "prod-category-nn-mv";
+    private static final String PRODUCT_CATEGORY_WITH_NAME_MATERIALIZED_VIEW = "prod-category-with-name-mv";
+    private static final String PRODUCT_CATEGORY_AGGREGATION_MATERIALIZED_VIEW = "prod-category-agg-mv";
+    private static final String PRODUCT_BRAND_CATEGORY_DETAIL_MATERIALIZED_VIEW = "prod-brand-category-detail-mv";
+    private static final String PRODUCT_ATTRIBUTE_MATERIALIZED_VIEW = "prod-attr-mv";
+    private static final String PRODUCT_ATTRIBUTE_VALUE_NONNULL_MATERIALIZED_VIEW = "prod-attr-value-nn-mv";
+    private static final String PRODUCT_ATTRIBUTE_VALUE_MATERIALIZED_VIEW = "prod-attr-value-mv";
+    private static final String PRODUCT_ATTRIBUTE_VALUE_WITH_NAME_MATERIALIZED_VIEW = "prod-attr-value-with-name-mv";
+    private static final String PRODUCT_ATTRIBUTE_AGGREGATION_MATERIALIZED_VIEW = "prod-attr-agg-mv";
+    private static final String PRODUCT_WITH_ALL_DETAIL_MATERIALIZED_VIEW = "product-with-all-detail-mv";
+
+    @Autowired
+    public ProductTopology(KafkaTopicConfig kafkaTopicConfig) {
+        super(kafkaTopicConfig);
+    }
+
+    @Override
+    @Autowired
+    protected void process(StreamsBuilder streamsBuilder) {
+        KTable<Long, ProductDto> productTable = createProductTable(streamsBuilder);
+        KTable<Long, BrandDto> brandTable = createBrandTable(streamsBuilder);
+        KTable<Long, ProductResultDto> productBrandTable = joinProductAndBrand(productTable, brandTable);
+
+        KTable<Long, CategoryDto> categoryTable = createCategoryTable(streamsBuilder);
+
+        KTable<Long, ProductCategoryDto> productCategoryTable =
+                createProductCategoryTable(streamsBuilder);
+
+        KTable<Long, ProductCategoryDto> nonNullProductCategoryTable =
```

Comment:
Could you explain, why we need to create a mv to hold `value:null` records here?